node/src/actors/chain_manager/handlers.rs

```diff
@@ -774,7 +767,22 @@ impl Handler<PeersBeacons> for ChainManager {
                     }
                 }
             }
+        };
+
+        if self.sm_state == StateMachine::Synchronizing {
+            if let Some(sync_start_epoch) = self.sync_waiting_for_add_blocks_since {
+                let current_epoch = self.current_epoch.unwrap();
+                let how_many_epochs_are_we_willing_to_wait_for_one_block_batch = 10;
```

Comment:
This is hardcoded to 10 epochs because it's a fix for when the node gets stuck, and that should never happen. We can use sentry to verify that it never happens, or decide a better timeout if it does happen to someone.